conan/api/subapi/profiles.py

```diff
@@ -62,10 +62,18 @@ def list(self):
         List all the profiles file sin the cache
         :return: an alphabetically ordered list of profile files in the default cache location
         """
+        # List is to be extended (directories should not have a trailing slash)
+        paths_to_ignore = ['.DS_Store']
+
         profiles = []
         profiles_path = self._cache.profiles_path
         if os.path.exists(profiles_path):
-            for current_directory, _, files in os.walk(profiles_path, followlinks=True):
+            for current_directory, directories, files in os.walk(profiles_path, followlinks=True):
+                directories[:] = [dir for dir in directories if os.path.relpath(
+                    os.path.join(current_directory, dir)) not in paths_to_ignore]
```

Comment:
I think this might not be necessary, a bit speculative of something that could happen in the future, but is not happening yet, as the ``.DS_store`` is always a file, never a folder. It is better in general to do the minimal changes to the code, and specially not add things that are actually not covered by tests.

Reply:
My bad I think I might overengineered it by stressing too much on the long-term scalability @RubenRBS was mentioning in the issue.

Reply:
No worries :)

Reply:
Yeah, don't worry, I was giving some feedback based on my own experience, but those things are completely subject to opinions and discussion. This is why I was saying that the PR is good, and my feedback was minor issues. Thanks!